sql: TestShowTenantFingerprintsProtectsTimestamp failed

sql.TestShowTenantFingerprintsProtectsTimestamp [failed](https://mesolite.cluster.engflow.com/invocations/default/458768df-9e63-4565-9fe9-e438aca17e16?testReportRun=1&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [c8d47d0123555cf64cfec6c55c29d0e8e63986eb](https://github.com/cockroachdb/cockroach/commits/c8d47d0123555cf64cfec6c55c29d0e8e63986eb):

Fatal error:

```
panic: test timed out after 15m0s
running tests:
	TestShowTenantFingerprintsProtectsTimestamp (14m7s)

```
Stack: 

```
goroutine 151408 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	pkg/server/node.go:1851
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func1
*   | 	pkg/rpc/context.go:713
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.ServerInterceptor.func12
*   | 	pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:92
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3
*   | 	pkg/rpc/context.go:165
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
*   | 	pkg/rpc/auth.go:90
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
*   | 	pkg/rpc/context.go:132
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
*   | 	pkg/rpc/context.go:130
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func2
*   | 	pkg/rpc/context.go:723
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewContext.ClientInterceptor.func8
*   | 	pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:232
*   | github.com/cockroachdb/cockroach/pkg/rpc.getChainUnaryInvoker.func1
*   | 	pkg/rpc/context.go:908
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func3
*   | 	pkg/rpc/context.go:793
*   | github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.Batch
*   | 	pkg/rpc/context.go:916
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).sendBatch
*   | 	pkg/kv/kvclient/kvcoord/transport.go:209
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).SendNext
*   | 	pkg/kv/kvclient/kvcoord/transport.go:187
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas
*   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:2747
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch
*   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:2203
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).divideAndSendBatchToRanges
*   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1738
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).Send
*   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1257
* Wraps: (2) close of closed channel
* Error types: (1) *withstack.withStack (2) runtime.plainError
*
    show_fingerprints_test.go:203: udating PTS reader cache to 1733328550.593914911,0
    show_fingerprints_test.go:215: enqueuing range 86 for mvccGC
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowTenantFingerprintsProtectsTimestamp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45210

sql.TestShowTenantFingerprintsProtectsTimestamp [failed](https://tanzanite.cluster.engflow.com/invocations/default/443b875b-c977-4fe3-abd3-6ff428b260d5?testReportRun=11&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [d7ea85402dc35e36c6cc35520fa91f25fd5c999d](https://github.com/cockroachdb/cockroach/commits/d7ea85402dc35e36c6cc35520fa91f25fd5c999d):

Fatal error:

```
panic: close of closed channel [recovered]
	panic: close of closed channel [recovered]
	panic: close of closed channel [recovered]
	panic: close of closed channel

```
Stack: 

```
goroutine 125814 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc00a1e2008, {0x8d29148, 0xc016ce3530}, {0x6d3a3c0?, 0xc0058797a0?})
	pkg/sql/conn_executor.go:1256 +0x19d
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1({0x8d29148, 0xc016ce3530}, {0xc00b169bc8?})
	pkg/sql/conn_executor.go:963 +0x45
panic({0x6d3a3c0?, 0xc0058797a0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	pkg/sql/colexecerror/error.go:137 +0x554
panic({0x6d3a3c0?, 0xc0058797a0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	pkg/sql/colexecerror/error.go:137 +0x554
panic({0x6d3a3c0?, 0xc0058797a0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.Wait({0xc0063246c0?, {0x8d29110?, 0xc011641a40?}, 0xc005a7c9f0?})
	pkg/util/ctxgroup/ctxgroup.go:150 +0x97
github.com/cockroachdb/cockroach/pkg/sql.(*planner).fingerprintSpanFanout(0xc00a1e26b8, {0x8d29148?, 0xc00e541020?}, {{0xc0079e4b98, 0x2, 0x8}, {0xc0079e4ba0, 0x2, 0x8}}, {0x0, ...}, ...)
	pkg/sql/fingerprint_span.go:196 +0x7a5
github.com/cockroachdb/cockroach/pkg/sql.(*planner).FingerprintSpan(0xc00a1e26b8, {0x8d29148?, 0xc01592f7d0?}, {{0xc0079e4b98, 0x2, 0x8}, {0xc0079e4ba0, 0x2, 0x8}}, {0x0, ...}, ...)
	pkg/sql/fingerprint_span.go:47 +0x165
github.com/cockroachdb/cockroach/pkg/sql.(*showFingerprintsNode).nextTenant(0xc0019be600, {{0x8d29148?, 0xc01592f7d0?}, 0xc006482808?, 0xc00a1e26b8?})
	pkg/sql/show_fingerprints.go:283 +0x474
github.com/cockroachdb/cockroach/pkg/sql.(*showFingerprintsNode).Next(0x651d4a0?, {{0x8d29148?, 0xc01592f7d0?}, 0xc006482808?, 0xc00a1e26b8?})
	pkg/sql/show_fingerprints.go:308 +0x4f2
github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Next(0xc00726cc08)
	pkg/sql/plan_node_to_row_source.go:211 +0xa8
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Next(0xc0021d6308)
	pkg/sql/colexec/columnarizer.go:234 +0x11f
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*invariantsChecker).Next(0xc0033021c0)
	pkg/sql/colexec/invariants_checker.go:88 +0x42
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).next(...)
	pkg/sql/colflow/stats.go:113
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc003302200?)
	pkg/sql/colexecerror/error.go:162 +0x57
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).Next(0xc0046ab700)
	pkg/sql/colflow/stats.go:121 +0x4c
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).nextAdapter(...)
	pkg/sql/colflow/flow_coordinator.go:241
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0x8d29110?)
	pkg/sql/colexecerror/error.go:162 +0x57
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).next(...)
	pkg/sql/colflow/flow_coordinator.go:245
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).Run(0xc00d361ea0, {0x8d29110?, 0xc0089b7400?})
	pkg/sql/colflow/flow_coordinator.go:277 +0x17f
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0xc005ac9640, {0x8d29110, 0xc0089b7400}, 0x0)
	pkg/sql/colflow/vectorized_flow.go:316 +0x1e6
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc006a0fa20, {0x8d29148, 0xc01592f200}, 0xc006159ba0, 0xc00ae5a280, 0xc0019be780, 0xc006c7fc08, 0xc00a1e2bc0, 0xc0088b9a08)
	pkg/sql/distsql_running.go:927 +0xd67
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc006a0fa20, {0x8d29148, 0xc01592f200}, 0xc00a1e2bc0, 0xc006159ba0, 0xc00ae5a280, {{0x8d271b8, 0xc0019be600}, 0x0}, 0xc006c7fc08, ...)
	pkg/sql/distsql_running.go:2032 +0x215
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll.func3(0x10?, 0x10?, {0x8d29148?, 0xc01592f200?}, 0x7f0524ee1f90?, 0xc0105e1210?, 0x0?)
	pkg/sql/distsql_running.go:1734 +0xbb
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll(0xc006a0fa20, {0x8d29148, 0xc01592f200}, 0xc00a1e2bc0, 0xc006159ba0, 0xc00a1e26b8, 0xc006c7fc08, 0x0)
	pkg/sql/distsql_running.go:1737 +0x229
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc00a1e2008, {0x8d29148, 0xc01592f200}, 0xc00a1e26b8, 0x3?, {0x7f0525ec5a98?, 0xc004c62f60?}, 0x0, 0xc00987b298, {0x8cc8560, ...})
	pkg/sql/conn_executor_exec.go:2435 +0x48a
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc00a1e2008, {0x8d29148, 0xc01592e870}, 0xc00a1e26b8, {0x7f0525ec5a98, 0xc004c62f60})
	pkg/sql/conn_executor_exec.go:1981 +0x119e
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc00a1e2008, {0x8d29148, 0xc01592e720}, {{0x8d4e3e0, 0xc005db5f40}, {0x0, 0x0, 0x0}, {0xc001da10d7, 0x3e}, ...}, ...)
	pkg/sql/conn_executor_exec.go:1187 +0x49c5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func1({0x8d29148?, 0xc01592e720?})
	pkg/sql/conn_executor_exec.go:141 +0xb0
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0xcff7900?, {0x8d29148?, 0xc01592e720?}, {0x8d4e3e0?, 0xc005db5f40?}, 0x1e?, 0xc004c62f60?)
	pkg/sql/conn_executor_exec.go:3451 +0x2ec
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc00a1e2008, {0x8d29148, 0xc01592e720}, {{0x8d4e3e0, 0xc005db5f40}, {0x0, 0x0, 0x0}, {0xc001da10d7, 0x3e}, ...}, ...)
	pkg/sql/conn_executor_exec.go:140 +0x725
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1(0xc00639f808, 0xc00639f8b8, 0xc00a1e2008, 0xc00639f4e0, 0xc00b169940, 0xc00b169930, 0xc00b169960, 0xc00639f978)
	pkg/sql/conn_executor.go:2349 +0x3f1
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc00a1e2008)
	pkg/sql/conn_executor.go:2354 +0x9f9
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc00a1e2008, {0x8d29148, 0xc016ce3530}, 0x3c?, 0xc008544380?, 0xc00ea29e70?)
	pkg/sql/conn_executor.go:2271 +0x208
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc008ade5a0?, {0x8d29148?, 0xc016ce3530?}, {0xc00de0a900?}, 0x2?, 0x0?)
	pkg/sql/conn_executor.go:965 +0xda
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommands(0xc004c62808, {0x8d29148, 0xc016ce3530}, {0x0, 0x4, {0x1, {0x72dc115, 0x3}, {0xc00556f350, 0xf}, ...}, ...}, ...)
	pkg/sql/pgwire/conn.go:251 +0x4c8
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl.func4()
	pkg/sql/pgwire/server.go:1196 +0xec
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl in goroutine 125768
	pkg/sql/pgwire/server.go:1193 +0x5db
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	pkg/kv/kvserver/store_send.go:188
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).SendWithWriteBytes
*   | 	pkg/kv/kvserver/stores.go:198
*   | github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal
*   | 	pkg/server/node.go:1650
*   | github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch
*   | 	pkg/server/node.go:1851
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func1
*   | 	pkg/rpc/context.go:713
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.ServerInterceptor.func12
*   | 	pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:92
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3
*   | 	pkg/rpc/context.go:165
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
*   | 	pkg/rpc/auth.go:90
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
*   | 	pkg/rpc/context.go:132
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
*   | 	pkg/rpc/context.go:130
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func2
*   | 	pkg/rpc/context.go:723
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewContext.ClientInterceptor.func8
*   | 	pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:232
*   | github.com/cockroachdb/cockroach/pkg/rpc.getChainUnaryInvoker.func1
*   | 	pkg/rpc/context.go:908
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func3
*   | 	pkg/rpc/context.go:793
*   | github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.Batch
*   | 	pkg/rpc/context.go:916
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).sendBatch
*   | 	pkg/kv/kvclient/kvcoord/transport.go:209
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).SendNext
*   | 	pkg/kv/kvclient/kvcoord/transport.go:187
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas
*   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:2755
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch
*   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:2211
* Wraps: (6) close of closed channel
* Error types: (1) *withstack.withStack (2) *safedetails.withSafeDetails (3) *withstack.withStack (4) *withstack.withStack (5) *withstack.withStack (6) runtime.plainError
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=11</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowTenantFingerprintsProtectsTimestamp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql.TestShowTenantFingerprintsProtectsTimestamp [failed](https://tanzanite.cluster.engflow.com/invocations/default/d5de3c34-bdb3-428f-af9f-1f856cc7e5fc?testReportRun=9&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [683618f4400ecbffb45564697fdb864909eca956](https://github.com/cockroachdb/cockroach/commits/683618f4400ecbffb45564697fdb864909eca956):

Fatal error:

```
panic: close of closed channel [recovered]
	panic: close of closed channel [recovered]
	panic: close of closed channel [recovered]
	panic: close of closed channel

```
Stack: 

```
goroutine 117921 [running]:
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc007e6c488, {0x8d29148, 0xc010818930}, {0x6d3a3c0?, 0xc00ec34ba0?})
	pkg/sql/conn_executor.go:1256 +0x19d
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1({0x8d29148, 0xc010818930}, {0xc00e651bc8?})
	pkg/sql/conn_executor.go:963 +0x45
panic({0x6d3a3c0?, 0xc00ec34ba0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	pkg/sql/colexecerror/error.go:137 +0x554
panic({0x6d3a3c0?, 0xc00ec34ba0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	pkg/sql/colexecerror/error.go:137 +0x554
panic({0x6d3a3c0?, 0xc00ec34ba0?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.Wait({0xc006ed8780?, {0x8d29110?, 0xc011d9fc20?}, 0xc00c394150?})
	pkg/util/ctxgroup/ctxgroup.go:150 +0x97
github.com/cockroachdb/cockroach/pkg/sql.(*planner).fingerprintSpanFanout(0xc007e6cb38, {0x8d29148?, 0xc011979dd0?}, {{0xc0079e3460, 0x2, 0x8}, {0xc0079e3468, 0x2, 0x8}}, {0x0, ...}, ...)
	pkg/sql/fingerprint_span.go:196 +0x7a5
github.com/cockroachdb/cockroach/pkg/sql.(*planner).FingerprintSpan(0xc007e6cb38, {0x8d29148?, 0xc0019a5920?}, {{0xc0079e3460, 0x2, 0x8}, {0xc0079e3468, 0x2, 0x8}}, {0x0, ...}, ...)
	pkg/sql/fingerprint_span.go:47 +0x165
github.com/cockroachdb/cockroach/pkg/sql.(*showFingerprintsNode).nextTenant(0xc001b45c80, {{0x8d29148?, 0xc0019a5920?}, 0xc0035af008?, 0xc007e6cb38?})
	pkg/sql/show_fingerprints.go:283 +0x474
github.com/cockroachdb/cockroach/pkg/sql.(*showFingerprintsNode).Next(0x651d4a0?, {{0x8d29148?, 0xc0019a5920?}, 0xc0035af008?, 0xc007e6cb38?})
	pkg/sql/show_fingerprints.go:308 +0x4f2
github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Next(0xc005202f08)
	pkg/sql/plan_node_to_row_source.go:211 +0xa8
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Next(0xc00ed78308)
	pkg/sql/colexec/columnarizer.go:234 +0x11f
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*invariantsChecker).Next(0xc0167de840)
	pkg/sql/colexec/invariants_checker.go:88 +0x42
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).next(...)
	pkg/sql/colflow/stats.go:113
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc0167de880?)
	pkg/sql/colexecerror/error.go:162 +0x57
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).Next(0xc00e9d2e00)
	pkg/sql/colflow/stats.go:121 +0x4c
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).nextAdapter(...)
	pkg/sql/colflow/flow_coordinator.go:241
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0x8d29110?)
	pkg/sql/colexecerror/error.go:162 +0x57
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).next(...)
	pkg/sql/colflow/flow_coordinator.go:245
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).Run(0xc013bc3e00, {0x8d29110?, 0xc0137dd270?})
	pkg/sql/colflow/flow_coordinator.go:277 +0x17f
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run(0xc004c03000, {0x8d29110, 0xc0137dd270}, 0x0)
	pkg/sql/colflow/vectorized_flow.go:316 +0x1e6
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc00e3bd080, {0x8d29148, 0xc0019a5560}, 0xc00b73e410, 0xc014ecfc20, 0xc001b45f00, 0xc00a2fdc08, 0xc007e6d040, 0xc00624fa08)
	pkg/sql/distsql_running.go:927 +0xd67
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc00e3bd080, {0x8d29148, 0xc0019a5560}, 0xc007e6d040, 0xc00b73e410, 0xc014ecfc20, {{0x8d271b8, 0xc001b45c80}, 0x0}, 0xc00a2fdc08, ...)
	pkg/sql/distsql_running.go:2032 +0x215
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll.func3(0x10?, 0x10?, {0x8d29148?, 0xc0019a5560?}, 0x7f4d76adff28?, 0xc0153e1ed0?, 0x0?)
	pkg/sql/distsql_running.go:1734 +0xbb
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll(0xc00e3bd080, {0x8d29148, 0xc0019a5560}, 0xc007e6d040, 0xc00b73e410, 0xc007e6cb38, 0xc00a2fdc08, 0x0)
	pkg/sql/distsql_running.go:1737 +0x229
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc007e6c488, {0x8d29148, 0xc0019a5560}, 0xc007e6cb38, 0x3?, {0x7f4d7fd3f4c0?, 0xc006533f60?}, 0x0, 0xc012465418, {0x8cc8560, ...})
	pkg/sql/conn_executor_exec.go:2435 +0x48a
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc007e6c488, {0x8d29148, 0xc0019a5170}, 0xc007e6cb38, {0x7f4d7fd3f4c0, 0xc006533f60})
	pkg/sql/conn_executor_exec.go:1981 +0x119e
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc007e6c488, {0x8d29148, 0xc0019a4cf0}, {{0x8d4e3e0, 0xc0167de340}, {0x0, 0x0, 0x0}, {0xc005a8d0d7, 0x3e}, ...}, ...)
	pkg/sql/conn_executor_exec.go:1187 +0x49c5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func1({0x8d29148?, 0xc0019a4cf0?})
	pkg/sql/conn_executor_exec.go:141 +0xb0
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0xcff7900?, {0x8d29148?, 0xc0019a4cf0?}, {0x8d4e3e0?, 0xc0167de340?}, 0x1e?, 0xc006533f60?)
	pkg/sql/conn_executor_exec.go:3451 +0x2ec
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc007e6c488, {0x8d29148, 0xc0019a4cf0}, {{0x8d4e3e0, 0xc0167de340}, {0x0, 0x0, 0x0}, {0xc005a8d0d7, 0x3e}, ...}, ...)
	pkg/sql/conn_executor_exec.go:140 +0x725
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1(0xc0075b5808, 0xc0075b58b8, 0xc007e6c488, 0xc0075b54e0, 0xc00e651940, 0xc00e651930, 0xc00e651960, 0xc0075b5978)
	pkg/sql/conn_executor.go:2349 +0x3f1
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc007e6c488)
	pkg/sql/conn_executor.go:2354 +0x9f9
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc007e6c488, {0x8d29148, 0xc010818930}, 0x3c?, 0xc014236a80?, 0xc011e545a0?)
	pkg/sql/conn_executor.go:2271 +0x208
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc00e42fc20?, {0x8d29148?, 0xc010818930?}, {0xc0047530e0?}, 0x2?, 0x0?)
	pkg/sql/conn_executor.go:965 +0xda
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommands(0xc006533808, {0x8d29148, 0xc010818930}, {0x0, 0x4, {0x1, {0x72dc115, 0x3}, {0xc0043f8220, 0xf}, ...}, ...}, ...)
	pkg/sql/pgwire/conn.go:251 +0x4c8
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl.func4()
	pkg/sql/pgwire/server.go:1196 +0xec
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl in goroutine 117940
	pkg/sql/pgwire/server.go:1193 +0x5db
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	pkg/kv/kvserver/store_send.go:188
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).SendWithWriteBytes
*   | 	pkg/kv/kvserver/stores.go:198
*   | github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal
*   | 	pkg/server/node.go:1650
*   | github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch
*   | 	pkg/server/node.go:1851
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func1
*   | 	pkg/rpc/context.go:713
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.ServerInterceptor.func12
*   | 	pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:92
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3
*   | 	pkg/rpc/context.go:165
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
*   | 	pkg/rpc/auth.go:90
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
*   | 	pkg/rpc/context.go:132
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
*   | 	pkg/rpc/context.go:130
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func2
*   | 	pkg/rpc/context.go:723
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewContext.ClientInterceptor.func8
*   | 	pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:232
*   | github.com/cockroachdb/cockroach/pkg/rpc.getChainUnaryInvoker.func1
*   | 	pkg/rpc/context.go:908
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func3
*   | 	pkg/rpc/context.go:793
*   | github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.Batch
*   | 	pkg/rpc/context.go:916
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).sendBatch
*   | 	pkg/kv/kvclient/kvcoord/transport.go:209
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).SendNext
*   | 	pkg/kv/kvclient/kvcoord/transport.go:187
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas
*   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:2755
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch
*   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:2211
* Wraps: (6) close of closed channel
* Error types: (1) *withstack.withStack (2) *safedetails.withSafeDetails (3) *withstack.withStack (4) *withstack.withStack (5) *withstack.withStack (6) runtime.plainError
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowTenantFingerprintsProtectsTimestamp.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/show_fingerprints_test.go
+++ b/pkg/sql/show_fingerprints_test.go
@@ -149,8 +149,7 @@ func TestShowTenantFingerprintsProtectsTimestamp(t *testing.T) {
 	testingRequestFilter := func(_ context.Context, ba *kvpb.BatchRequest) *kvpb.Error {
 		for _, req := range ba.Requests {
 			if expReq := req.GetExport(); expReq != nil {
-				if expReq.ExportFingerprint && !exportStartedClosed.Load() {
-					exportStartedClosed.Store(true)
+				if expReq.ExportFingerprint && exportStartedClosed.CompareAndSwap(false, true) {
 					close(exportsStarted)
 					<-exportsResume
 				}

